src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -36,10 +37,38 @@ public class Shredder {
 
   private final DocumentLimitsConfig documentLimits;
 
+  private final JsonShreddingMetricsReporter jsonShreddingMetricsReporter;
+
   @Inject
-  public Shredder(ObjectMapper objectMapper, DocumentLimitsConfig documentLimits) {
+  public Shredder(
+      ObjectMapper objectMapper,
+      DocumentLimitsConfig documentLimits,
+      JsonShreddingMetricsReporter jsonShreddingMetricsReporter) {
```

Comment:
I think that what needs to be injected is `JsonShreddingMetricsFactory`: and then shredder will hold a reference to that factory. Factory's reporter-creating factory method is then called when reporter instance is needed, later on.